lark/lark.py

```diff
@@ -358,10 +363,10 @@ def _load(self, f, transformer=None, postlex=None):
         assert memo
         memo = SerializeMemoizer.deserialize(memo, {'Rule': Rule, 'TerminalDef': TerminalDef}, {})
         options = dict(data['options'])
-        if transformer is not None:
-            options['transformer'] = transformer
-        if postlex is not None:
-            options['postlex'] = postlex
+        if _LOAD_BLOCKED_OPTIONS.intersection(kwargs.keys()):
```

Comment:
You can remove `.keys()` in the previous line.  A `dict` when used as an iterable is its keys!

Reply:
I made the conscious decisions to use `.keys()`: explicit is better than implicit. But maybe my judgement was wrong on that. Lets hear @erezsh as a third opinion.

Reply:
I don't have a strong opinion about that. But personally I'd prefer

```python
_LOAD_BLOCKED_OPTIONS & set(...)
```

Over intersection.

Reply:
Also, why do you even need `_LOAD_BLOCKED_OPTIONS`? Why not just

```python
if set(kwargs) - _LOAD_ALLOWED_OPTIONS:
   # raise error
```

Reply:
And why do we need `temp_options` if we have `kwargs`?

Reply:
temp_options to deal with cache. If we used cache, the illegal option of cache will get passed through, but I will take a look.

> Also, why do you even need _LOAD_BLOCKED_OPTIONS

I am not that proficient with set arithmetic, yes you are correct both suggestions are better.

Reply:
Aha. One thing: if we `set(kwargs) - _LOAD_ALLOWED_OPTIONS` completely wrong options will get the same error message as for options that normally can be set. That is what I was avoiding with `_LOAD_BLOCKED_OPTIONS`. There are a lot of ways on how to solve this, this just seemed easiest.

Reply:
Well, you can change the error message based on if it's `in LarkOptions._defaults`

I agree it's not a huge difference, but it feels better to me.

Reply:
Removing temp_options on the other hand is quite the pain. We need to filter out the arguments that can be passed to `Lark`, but not to `Load._lark`, but which still make sense when using the cache, e.g. the entire `_LOAD_BLOCKED_OPTIONS`. I will push a commit, but I am not sure how nice this is.

Reply:
I don't understand why it's a pain. The standalone parser isn't affected by `cache` anyway. (it can be assumed to be always on regardless)

Reply:
Because the same method (`_load`) is used. Changes to the system how the standalone parser operators also have to be reflected in the cache handling in some way. This is one of them. View the latest commit. But it is less of a pain than I thought.